core/ledger/kvledger/kv_ledger.go

```diff
@@ -148,7 +149,11 @@ func newKVLedger(initializer *lgrInitializer) (*kvLedger, error) {
 	logger.Debugf("Register state db for chaincode lifecycle events: %t", ccEventListener != nil)
 	if ccEventListener != nil {
 		cceventmgmt.GetMgr().Register(ledgerID, ccEventListener)
-		initializer.ccLifecycleEventProvider.RegisterListener(ledgerID, &ccEventListenerAdaptor{ccEventListener})
+		initializer.ccLifecycleEventProvider.RegisterListener(
```

Comment:
Now, the register function is doing more than just registering. However, I could not think of any other simple way as the existing dependencies are complex already.

(1) lifecycle register a listener with the ledger to receive state updates whereas the ledger register a listener with lifecycle to receive the CouchDB index
(2) lifecycle cache needs to be initialized from the ledger and the index are initialized from the lifecycle 

As it requires a larger refactoring, I am okay with this.